src/sql/src/plan/statement.rs

```diff
@@ -578,14 +584,37 @@ fn handle_create_sink(
     let as_of = as_of.map(|e| query::eval_as_of(scx, e)).transpose()?;
     let connector_builder = match connector {
         Connector::File { .. } => unsupported!("file sinks"),
-        Connector::Kafka { broker, topic } => kafka_sink_builder(
-            format,
-            with_options,
+        Connector::Kafka {
             broker,
             topic,
-            from.desc()?.clone(),
-            suffix,
-        )?,
+            keys,
+        } => {
+            let desc = from.desc()?;
+            let key_indices = if let Some(keys) = keys {
+                let idces = keys
+                    .iter()
+                    .map(|k| {
+                        desc.iter_names()
+                            .position(|name| {
+                                name.map(|cn| cn.as_str() == k.as_str()).unwrap_or(false)
+                            })
+                            .ok_or_else(|| anyhow!("No such column: {}", k))
```

Comment:
I can't tell between this and the `collect` whether you need a `bail!` instead of an `anyhow!` to exit on error. I just know that I've messed that up multiple times :)

Reply:
`bail!(...)` is just sugar for `return Err(anyhow!(...))`. Since the `anyhow!` is in a closure, they're equivalent.